StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CommentsMustContainText.cs

```diff
@@ -89,7 +90,9 @@ private static void HandleMultiLineComment(SyntaxTreeAnalysisContext context, Sy
         private static void HandleSingleLineComment(SyntaxTreeAnalysisContext context, SyntaxTrivia singleLineComment)
         {
             int index = 0;
-            var list = TriviaHelper.GetContainingTriviaList(singleLineComment, out index);
+
+            // PERF: Explicitly cast to IReadOnlyList so we only box once.
+            IReadOnlyList<SyntaxTrivia> list = TriviaHelper.GetContainingTriviaList(singleLineComment, out index);
```

Comment:
:bulb: Boxing can completely eliminated by changing the signature of the following methods in `TriviaHelper`:

```csharp
internal static int IndexOfFirstNonWhitespaceTrivia<T>(T triviaList, bool endOfLineIsWhitespace = true)
    where T : IReadOnlyList<SyntaxTrivia>

internal static int IndexOfTrailingWhitespace<T>(T triviaList)
    where T : IReadOnlyList<SyntaxTrivia>
```